packages/components/src/navigation/stories/index.js

```diff
@@ -1,88 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import Navigation from '../';
+import NavigationBackButton from '../back-button';
+import NavigationMenu from '../menu';
+import NavigationTitle from '../title';
 
 export default {
 	title: 'Components/Navigation',
 	component: Navigation,
 };
 
 const data = [
-	{ title: 'WooCommerce', slug: 'root', back: 'Dashboard' },
-	{ title: 'Home', slug: 'home', parent: 'root', menu: 'primary' },
+	{ title: 'Home', id: 'home' },
 	{
 		title: 'Analytics',
-		slug: 'analytics',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'analytics',
 	},
 	{
 		title: 'Orders',
-		slug: 'orders',
-		parent: 'root',
-		back: 'WooCommerce Home',
-		menu: 'primary',
+		id: 'orders',
 	},
 	{
 		title: 'Overview',
-		slug: 'overview',
+		id: 'overview',
 		parent: 'analytics',
 	},
 	{
 		title: 'Products report',
-		slug: 'products',
+		id: 'products',
 		parent: 'analytics',
 	},
 	{
 		title: 'All orders',
-		slug: 'all_orders',
+		id: 'all_orders',
 		parent: 'orders',
 	},
 	{
 		title: 'Payouts',
-		slug: 'payouts',
+		id: 'payouts',
 		parent: 'orders',
 	},
 	{
 		title: 'Settings',
-		slug: 'settings',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'settings',
 		menu: 'secondary',
 	},
 	{
 		title: 'Extensions',
-		slug: 'extensions',
-		parent: 'root',
-		back: 'WooCommerce Home',
+		id: 'extensions',
 		menu: 'secondary',
 	},
 	{
 		title: 'General',
-		slug: 'general',
+		id: 'general',
 		parent: 'settings',
 	},
 	{
 		title: 'Tax',
-		slug: 'tax',
+		id: 'tax',
 		parent: 'settings',
 	},
 	{
 		title: 'My extensions',
-		slug: 'my_extensions',
+		id: 'my_extensions',
 		parent: 'extensions',
 	},
 	{
 		title: 'Marketplace',
-		slug: 'marketplace',
+		id: 'marketplace',
 		parent: 'extensions',
 	},
 ];
 
 function Example() {
-	return <Navigation data={ data } initial="home" />;
+	const [ active, setActive ] = useState( 'home' );
+	const items = data.map( ( item ) => {
+		item.onClick = () => setActive( item.id );
+		return item;
+	} );
```

Comment:
This is an awkward API. I think it would be better to leave data as purely data. Instead of asking users to manipulate their data, can we just supply a prop for a callback?

Reply:
This is not part of the API, this is part of the user supplied data.  It may look odd because of our `Example` setup, but this is just supplying the `onClick` behavior as data:

```
[
  {
    id: 'item-1'.
    onClick: () => setActive( 'item-1' ),
  }
]
```